cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/StrimziPodSetController.java

```diff
@@ -82,58 +82,63 @@ public class StrimziPodSetController implements Runnable {
-                enqueueStrimziPodSet(newPodSet, "MODIFIED");
-            }
+                @Override
+                public void onUpdate(StrimziPodSet oldPodSet, StrimziPodSet newPodSet) {
+                    enqueueStrimziPodSet(newPodSet, "MODIFIED");
+                }
 
-            @Override
-            public void onDelete(StrimziPodSet podSet, boolean deletedFinalStateUnknown) {
-                LOGGER.debugOp("StrimziPodSet {} in namespace {} was {}", podSet.getMetadata().getName(), podSet.getMetadata().getNamespace(), "DELETED");
-                // Nothing to do => garbage collection should take care of things
-            }
-        }, 10 * 60 * 1000);
-
-        // Pod informer and lister is used to get events about pods and get pods quickly
-        this.podInformer = podOperator.informer(watchedNamespace, Map.of(Labels.STRIMZI_KIND_LABEL, "Kafka"));
-        this.podLister = new Lister<>(podInformer.getIndexer());
-        this.podInformer.addEventHandlerWithResyncPeriod(new ResourceEventHandler<>() {
-            @Override
-            public void onAdd(Pod pod) {
-                enqueuePod(pod, "ADDED");
-            }
+                @Override
+                public void onDelete(StrimziPodSet podSet, boolean deletedFinalStateUnknown) {
+                    LOGGER.debugOp("StrimziPodSet {} in namespace {} was {}", podSet.getMetadata().getName(), podSet.getMetadata().getNamespace(), "DELETED");
+                    // Nothing to do => garbage collection should take care of things
+                }
+            }, 10 * 60 * 1000);
+
+            // Pod informer and lister is used to get events about pods and get pods quickly
+            this.podInformer = podOperator.informer(watchedNamespace, Map.of(Labels.STRIMZI_KIND_LABEL, "Kafka"));
+            this.podLister = new Lister<>(podInformer.getIndexer());
+            this.podInformer.addEventHandlerWithResyncPeriod(new ResourceEventHandler<>() {
+                @Override
+                public void onAdd(Pod pod) {
+                    enqueuePod(pod, "ADDED");
+                }
 
-            @Override
-            public void onUpdate(Pod oldPod, Pod newPod) {
-                enqueuePod(newPod, "MODIFIED");
-            }
+                @Override
+                public void onUpdate(Pod oldPod, Pod newPod) {
+                    enqueuePod(newPod, "MODIFIED");
+                }
 
-            @Override
-            public void onDelete(Pod pod, boolean deletedFinalStateUnknown) {
-                enqueuePod(pod, "DELETED");
-            }
-        }, 10 * 60 * 1000);
+                @Override
+                public void onDelete(Pod pod, boolean deletedFinalStateUnknown) {
+                    enqueuePod(pod, "DELETED");
+                }
+            }, 10 * 60 * 1000);
 
-        controllerThread = new Thread(this, "StrimziPodSetController");
+            controllerThread = new Thread(this, "StrimziPodSetController");
+        } catch (Throwable e) {
```

Comment:
Why is this needed?

Reply:
That way I can print an error log which tells the user that the OP failed during initialization, without having to look at the stack to figure it out.

Reply:
Maybe you can wrap it on the outside then instead of doing this in the constructor? I do not really like the idea of catching something here just to not do anything and just log it.

Reply:
Yes, much better. Thanks.